mobile-widgets/qmlmanager.cpp

```diff
@@ -564,9 +578,7 @@ void QMLManager::loadDivesWithValidCredentials()
 	QString url;
 	timestamp_t currentDiveTimestamp = m_selectedDiveTimestamp;
 	if (getCloudURL(url)) {
-		QString errorString(get_error_string());
-		appendTextToLog(errorString);
-		setStartPageText(RED_FONT + tr("Cloud storage error: %1").arg(errorString) + END_FONT);
+		setStartPageText(RED_FONT + tr("Cloud storage error: %1").arg(m_lastError) + END_FONT);
```

Comment:
AFAIK, it's in the same thread, but do we know for sure that `m_lastError` will be related to `loadDivesWithValidCredentials()` at all times here and not being an error for something else?

Reply:
similar comment applies to other usages of `m_lastError`.

Reply:
Yes, but the code had the same issue before. Before, it would show more than one message, now it shows the last message. We could accumulate messages in `m_lastError`, but then we could remove the corresponding logic from `report_error()`. That would probably be more robust, since `QString` is thread safe AFAIK. In any case, the fundamental problem of message-mixing remains.

The whole `report_error()`/`get_error_string()` thing is scarily thread unsafe. I think we had a short discussion on the mailing list in which it was concluded with that this should never happen.

I see no problem making it thread safe (without introducing C++isms into the C part), btw. Should I go for it?

Reply:
yep, i know it was the same issue as before. i haven't seen this discussion on the ML, but child threads should simply not call `report_error()`.

if it was already discussed that it should never happen, we can just leave it as is.